day_7/10_speckled_frogs.rb

```diff
@@ -0,0 +1,149 @@
+
+# use frogs count from 10 to 1
+for frogs in (10).downto(1)
+  case frogs
+  # Print when frogs is 10
+  when 10
+    puts "#{(in_words(frogs)).capitalize} speckled frogs sat on a log eating some most delicious bugs."
+    puts "One jumped in the pool where its nice and cool, then there were #{in_words(frogs-1)} frogs."
+    frogs -= 1
+  # Print when frogs is 7 to 9
+  when 8..9
+    puts "#{frogs} speckled frogs sat on a log eating some most delicious bugs."
+    puts "One jumped in the pool where its nice and cool, then there were #{frogs-1} frogs."
+    frogs -= 1
+  # Print when frogs is 6
+  when 7
+    puts "#{(in_words(frogs)).capitalize} speckled frogs sat on a log eating some most delicious bugs."
+    puts "One jumped in the pool where its nice and cool, then there were #{in_words(frogs-1)} frogs."
+    frogs -= 1
+  # Print when frogs is 6
+  when 2..6
+    puts "#{frogs} speckled frogs sat on a log eating some most delicious bugs."
+    puts "One jumped in the pool where its nice and cool, then there were #{frogs-1} frogs."
+    frogs -= 1
+  # Print when frogs reaches end (1)
+  else
+    puts "#{frogs} speckled frog sat on a log eating some most delicious bugs."
+    puts "One jumpted in the pool where its nice and cool,"
+    puts "then there were no more speckled frogs!"
+  end
+end
+
+puts "=====" * 10
+
+
+# Extension 2
+puts "-----" * 10
+puts "10 Speckled Frogs - Extension 2"
+puts "-----" * 10
+puts "How many frogs are on the log?"
+print "> "
+frogs_start = ($stdin.gets.chomp).to_i
+
+# use frogs count from 10 to 1
+for frogs in ((frogs_start)).downto(1)
+  case frogs
+  # Print when frogs is 10
+  when (frogs_start)
+    puts "#{(in_words(frogs)).capitalize} speckled frogs sat on a log eating some most delicious bugs."
+    puts "One jumped in the pool where its nice and cool, then there were #{in_words(frogs-1)} frogs."
+    frogs -= 1
+  # Print when frogs is frogs_start between frogs_start-3 to -1
+when (frogs_start-2)..(frogs_start-1)
+    puts "#{frogs} speckled frogs sat on a log eating some most delicious bugs."
+    puts "One jumped in the pool where its nice and cool, then there were #{frogs-1} frogs."
+    frogs -= 1
+  # Print when frogs is 6
+when (frogs_start-3)
+    puts "#{(in_words(frogs)).capitalize} speckled frogs sat on a log eating some most delicious bugs."
+    puts "One jumped in the pool where its nice and cool, then there were #{in_words(frogs-1)} frogs."
+    frogs -= 1
+  # Print when frogs is 6
+when 2..(frogs_start-4)
+    puts "#{frogs} speckled frogs sat on a log eating some most delicious bugs."
+    puts "One jumped in the pool where its nice and cool, then there were #{frogs-1} frogs."
+    frogs -= 1
+  # Print when frogs reaches end (1)
+  else
+    puts "#{frogs} speckled frog sat on a log eating some most delicious bugs."
+    puts "One jumpted in the pool where its nice and cool,"
+    puts "then there were no more speckled frogs!"
+  end
+end
+
+puts "=====" * 10
```

Comment:
Awesome effort here, tackling a variety of extensions! One idea to clean this up a tad is to store repeated lines in variables... anytime you see repeats of strings, or logic, try to break that out into a separate variable or function for reusability!